spec/parallel_tests/grouper_spec.rb

```diff
@@ -76,50 +76,116 @@ def call(num_groups, options = {})
+        )
+      end
 
-    it "specify_groups aborts when spec passed in doesn't match existing specs" do
-      expect do
-        call(3, specify_groups: '1|2|6')
-      end.to raise_error(
-        "Could not find [\"6\"] from --specify-groups in the selected files & folders"
-      )
-    end
+      it "specify_groups aborts when spec passed in doesn't match existing specs" do
+        expect do
+          call(3, specify_groups: '1|2|6')
+        end.to raise_error(
+          "Could not find [\"6\"] from --specify-groups in the selected files & folders"
+        )
+      end
 
-    it "specify_groups aborts when spec passed in doesn't match existing specs again" do
-      expect do
-        call(3, specify_groups: '1,6|2')
-      end.to raise_error(
-        "Could not find [\"6\"] from --specify-groups in the selected files & folders"
-      )
-    end
+      it "specify_groups aborts when spec passed in doesn't match existing specs again" do
+        expect do
+          call(3, specify_groups: '1,6|2')
+        end.to raise_error(
+          "Could not find [\"6\"] from --specify-groups in the selected files & folders"
+        )
+      end
 
-    it "specify_groups aborts when number of specs is equal to number passed in" do
-      expect do
-        call(3, specify_groups: '1|2|3')
-      end.to raise_error(/The specs that aren't run:\n\["4", "5"\]/)
+      it "specify_groups aborts when number of specs is equal to number passed in" do
+        expect do
+          call(3, specify_groups: '1|2|3')
+        end.to raise_error(/The specs that aren't run:\n\["4", "5"\]/)
+      end
+
+      it "specify_groups does not abort when the every single spec is specified in it" do
+        expect(call(3, specify_groups: '1,2|3,4|5')).to eq([["1", "2"], ["3", "4"], ["5"]])
+      end
     end
 
-    it "specify_groups does not abort when the every single spec is specified in it" do
-      expect(call(3, specify_groups: '1,2|3,4|5')).to eq([["1", "2"], ["3", "4"], ["5"]])
+    context 'with specific groups provided through STDIN' do
+      it "groups specify_groups as specified when specify_groups is just one spec" do
+        allow($stdin).to receive(:gets).and_return('1')
+
+        expect(call(3, specify_groups: '-')).to eq([["1"], ["2", "5"], ["3", "4"]])
+      end
+
+      it "groups specify_groups as specified when specify_groups is just multiple specs in one process" do
+        allow($stdin).to receive(:gets).and_return('3,1')
+
+        expect(call(3, specify_groups: '-')).to eq([["3", "1"], ["5"], ["2", "4"]])
+      end
+
+      it "groups specify_groups as specified when specify_groups is multiple specs" do
+        allow($stdin).to receive(:gets).and_return('1,2|4')
+
+        expect(call(3, specify_groups: '-')).to eq([["1", "2"], ["4"], ["3", "5"]])
+      end
+
+      it "specify_groups aborts when number of specs separated by pipe is out of bounds" do
+        allow($stdin).to receive(:gets).and_return('1|2|3|4')
```

Comment:
without `chomp: true`, `gets` will return the line terminator.

https://rubyapi.org/3.4/o/io#method-i-gets